rofi-pass

```diff
@@ -79,22 +79,24 @@ autopass () {
   rm -f "$HOME/.cache/rofi-pass/last_used"
   printf '%s\n' "${root}: $selected_password" > "$HOME/.cache/rofi-pass/last_used"
   for word in ${stuff["$AUTOTYPE_field"]}; do
-  case "$word" in
-    ":tab") xdotool key Tab;;
-    ":space") xdotool key space;;
-    ":delay") sleep "${delay}";;
-    ":enter") xdotool key Return;;
-    ":otp") printf '%s' "$(generateOTP)" | xdotool type --clearmodifiers --file -;;
-    "pass") printf '%s' "${password}" | xdotool type --clearmodifiers --file -;;
-    ":clip") inclip=true ;;
-    *)
-        if [[ ${inclip} == "true" ]]; then
-            clipcontent=$(printf '%s%s' "$clipcontent" "${stuff[${word}]} ")
-        else
-            printf '%s' "${stuff[${word}]}" | xdotool type --clearmodifiers --file -
-        fi
-    ;;
-  esac
+    wordToPrint=""
+    case "$word" in
+      ":tab") xdotool key Tab;;
+      ":space") xdotool key space;;
+      ":delay") sleep "${delay}";;
+      ":enter") xdotool key Return;;
+      ":otp") wordToPrint="$(generateOTP)";;
+      "pass") wordToPrint="${password}";;
+      ":clip") inclip=true;;
+      *) wordToPrint="${stuff[${word}]}";;
+    esac
+    if [[ ! -z "$wordToPrint" ]]; then
+      if [[ ${inclip} == "true" ]]; then
+        clipcontent=$(printf '%s%s ' "$clipcontent" "$wordToPrint")
```

Comment:
What is the `'%s%s '` supposed to be? Why is there a whitespace at the end?

Reply:
`'%s%s '` basically requires two string arguments. (in that case clipcontent wordToPrint).

I don't know if the whitespace is required but lets look at an example:

If we have:

`:clip user pass`

without the whitespace the clipboard content after the auto fill would look like: `userpass`. Which is probably not the expected behaviour, therefore there's a whitespace after it so it turns to `user pass`.

Reply:
Certainly you mean `%s %s` instead of `%s%s `?

Reply:
Let's say we have this auto pass field: `:clip user otp url`

Then the content of the clipcontent looks like this (loop iteration -> result):

Look at the whitespaces here:

* clipcontent=""
* `clipcontent=$(printf '%s%s ' "" "<user>")`  -> `result: clipcontent="<user> "`
* `clipcontent=$(printf '%s%s ' "<user> " "<otp>")` ->` result: clipcontent="<user> <otp> "`
* `clipcontent=$(printf '%s%s ' "<user> <otp> " "<url>")` -> `result: clipcontent="<user> <otp> <url> "`

Reply:
Does the whitespace at the end of the string not pose any issue?